src/compiler/checker.ts

```diff
@@ -9937,7 +9938,7 @@ namespace ts {
 
         // Return the contextual type for a given expression node. During overload resolution, a contextual type may temporarily
         // be "pushed" onto a node using the contextualType property.
-        function getApparentTypeOfContextualType(node: Expression): Type {
+        function getApparentTypeOfContextualType(node: Expression): ObjectType {

```

Comment:
The returned type could be something other than an `ObjectType` so this change isn't correct.